[Bug]: installation through Makefile fails because of missing 'packaging' dependency

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

During ADF installation using the new method using Makefile, I get a problem of missing dependency.

### Expected Behavior

Install using make command should build the SAM artifacts and start the deployment

### Current Behavior

Using a Cloud9 instance, launching the `make` command after a while I'm getting this error:

```
Generate a new version number...
( \
        . .venv/bin/activate; \
        BASE_ADF_VERSION=$(cat ./src/template.yml | yq '.Metadata."AWS::ServerlessRepo::Application".SemanticVersion' -r); \
        COMMIT_ADF_VERSION=v4.0.0; \
        sed "s/Version: $BASE_ADF_VERSION/Version: $COMMIT_ADF_VERSION/g" ./src/template.yml > ./src/template-sam.yml; \
)
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ModuleNotFoundError: No module named 'packaging'

Error: ADF Main template version is newer than the requested git tag version!
Most likely, the git tags have not been fetched recently yet.

Please fetch the git tags from the cloned repository to continue.
You can do this by running: git fetch --tags origin

ADF Main template version (src/template.yml): v4.0.0
Resolved ADF version using git tags: v4.0.0

make: *** [Makefile:237: verify_version] Error 1
```

The error message is misleading, the real error isn't the version mismatch, but the missing 'packaging' module

### Steps To Reproduce

1. Create a Cloud9 instance
2. Open the IDE
3. From the terminal, clone the ADF repository
4. cd into the cloned folder
5. launch `make`


### Possible Solution

Add the 'packaging' module installation in the `build_deps` step

### Additional Information/Context

_No response_

### ADF Version

4.0.0

### Contributing a fix?

- [X] Yes, I am working on a fix to resolve this issue

--- a/Makefile
+++ b/Makefile
@@ -129,7 +129,7 @@ version_report: report_versions
 build_deps: .venv
 	( \
 		. .venv/bin/activate; \
-		pip install aws-sam-cli yq; \
+		pip install aws-sam-cli yq packaging; \
 	)
 
 deps: .venv

--- a/Makefile
+++ b/Makefile
@@ -2,7 +2,7 @@
 # SPDX-License-Identifier: Apache-2.0
 
 # Makefile versions
-MAKEFILE_VERSION := 2.3
+MAKEFILE_VERSION := 2.3.1
 UPDATE_VERSION := make/latest
 
 # This Makefile requires Python version 3.9 or later

